src/aioquic/quic/congestion/cubic.py

```diff
@@ -0,0 +1,210 @@
+from . import QuicCongestionControl, QuicRttMonitor, K_INITIAL_WINDOW, K_MINIMUM_WINDOW
+from ..packet_builder import QuicSentPacket
+from typing import Iterable, Optional, Dict, Any
+
+# cubic specific variables (see https://www.rfc-editor.org/rfc/rfc9438.html#name-definitions)
+K_CUBIC_K = 1    
+K_CUBIC_C = 0.4
+K_CUBIC_LOSS_REDUCTION_FACTOR = 0.7
+K_CUBIC_MAX_IDLE_TIME = 2   # reset the cwnd after 2 seconds of inactivity
+
+class CubicCongestionControl(QuicCongestionControl):
+    """
+    Cubic congestion control implementation for aioquic
+    """
+
+    def __init__(self, max_datagram_size : int, callback=None, reno_friendly_activated = True) -> None:
+        super().__init__(max_datagram_size=max_datagram_size, callback=callback)
+        self.additive_increase_factor = max_datagram_size  # increase by one segment
+        
+        self._max_datagram_size = max_datagram_size
+        self.congestion_window = K_INITIAL_WINDOW * self._max_datagram_size
+        self._congestion_recovery_start_time = 0.0
+        self.ssthresh: Optional[int] = None
+        self.reno_friendly_activated = reno_friendly_activated
+
+        self._rtt_monitor = QuicRttMonitor()
+
+        self.reset()
+
+        self.last_ack = None
+
+        self.now = 0
+
+    def better_cube_root(self, x):
+        if (x < 0):
+            # avoid precision errors that make the cube root returns an imaginary number
+            return -((-x)**(1./3.))
+        else:
+            return (x)**(1./3.)
+        
+    def W_cubic(self, t):
+        W_max_segments = self._W_max / self._max_datagram_size
+        target_segments = K_CUBIC_C * (t - self.K)**3 + (W_max_segments)
+        return target_segments * self._max_datagram_size
+    
+    def is_reno_friendly(self, t) -> bool:
+        return self.reno_friendly_activated and self.W_cubic(t) < self._W_est
+    
+    def is_concave(self):
+        return self.congestion_window < self._W_max
+    
+    def is_convex(self):
+        return self.congestion_window >= self._W_max
+    
+    def reset(self):
+        self.congestion_window = K_INITIAL_WINDOW * self._max_datagram_size
+
+        self._cwnd_prior = None
+        self._cwnd_epoch = None
+        self._t_epoch = None
+        self._W_max = None
+        self._first_slow_start = True
+        self._starting_congestion_avoidance = False
+        self.K = 0
+        self._W_est = 0
+        self._cwnd_epoch = 0
+        self._t_epoch = 0
+        self._W_max = self.congestion_window
+
+    def on_packet_acked(self, packet: QuicSentPacket, now: float) -> None:
+        super().on_packet_acked(packet, now)
+        rtt = self.recovery._rtt_smoothed
```

Comment:
No, the congestion control should not have a handle to the `QuicPacketRecovery`. Is it actually the smooth RTT we want here or the latest?

Reply:
It should be according to the RFC the weighted average RTT

Reply:
But I'm wondering if we couldn't use the latest RTT indeed ?

Reply:
I don't know the RFCs enough, if you need it add a `smoothed_rtt` keyword argument to `on_rtt_measurement`